migrations.go

```diff
@@ -261,15 +281,34 @@ func RunMigrations(db DB, migrations []Migration, a ...string) (oldVersion, newV
 			if m.Version <= oldVersion {
 				continue
 			}
-			err = m.Up(db)
-			if err != nil {
-				return
+
+			upFunc := func(db DB, m *Migration) error {
+				err = m.Up(db)
+				if err != nil {
+					return err
+				}
+				newVersion = m.Version
+				err = SetVersion(db, newVersion)
+				if err != nil {
+					return err
+				}
+				return nil
 			}
-			newVersion = m.Version
-			err = SetVersion(db, newVersion)
-			if err != nil {
-				return
+
+			if m.Transactional {
+				err = db.(*pg.DB).RunInTransaction(func(tx *pg.Tx) error {
```

Comment:
db can be `*pg.DB` or `*pg.Tx`. I think we should support both cases to preserve backwards compatibility.